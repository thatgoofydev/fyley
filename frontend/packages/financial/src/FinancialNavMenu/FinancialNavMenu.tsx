import React, { FunctionComponent, useState } from "react";
import { useHistory, useLocation } from "react-router-dom";
import { HeaderNavItem } from "@fyley/ui-lib";

enum FinancialNavOptions {
  Overview = "overview",
  Accounts = "accounts"
}

const useFinancialNav = (baseRoute: string, onItemClicked: () => void) => {
  const [currentOption, setOption] = useState<FinancialNavOptions>();
  const history = useHistory();
  const location = useLocation();

  const navigateTo = (route: FinancialNavOptions) => {
    history.push(`${baseRoute}/${route.toString()}`);
    onItemClicked();
    setOption(route);
  };

  const isActive = location.pathname.startsWith(baseRoute);

  return {
    currentOption,
    isActive,
    navigateTo
  };
};

interface IFinancialHeaderNavProps {
  baseRoute: string;
  open: boolean;
  onItemClicked: () => void;
}

export const FinancialHeaderNavItem: FunctionComponent<IFinancialHeaderNavProps> = ({
  baseRoute,
  open,
  onItemClicked
}) => {
  const { currentOption, isActive, navigateTo } = useFinancialNav(baseRoute, onItemClicked);
  const history = useHistory();
  const location = useLocation();

  const navigateToOverview = () => navigateTo(FinancialNavOptions.Overview);
  const navigateToAccounts = () => navigateTo(FinancialNavOptions.Accounts);

  return (
    <HeaderNavItem open={open} active={isActive} label="Financial" onItemClicked={onItemClicked}>
      <HeaderNavItem
        label={FinancialNavOptions.Overview}
        active={currentOption === FinancialNavOptions.Overview}
        onItemClicked={() => navigateTo(FinancialNavOptions.Overview)}
      />
      <HeaderNavItem
        label={FinancialNavOptions.Accounts}
        active={currentOption === FinancialNavOptions.Accounts}
        onItemClicked={() => navigateTo(FinancialNavOptions.Accounts)}
      />
    </HeaderNavItem>
  );
};
