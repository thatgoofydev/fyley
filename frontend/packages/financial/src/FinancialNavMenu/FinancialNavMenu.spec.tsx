import React from "react";
import { render, cleanup } from "@testing-library/react";
import { FinancialHeaderNavItem, IFinancialHeaderNavProps } from "./FinancialNavMenu";
import { MemoryRouter } from "react-router-dom";

const noop = () => {};
const defaultProps: IFinancialHeaderNavProps = {
  open: true,
  baseRoute: "/financial",
  onItemClicked: noop
};

afterEach(cleanup);

describe("FinancialNavMenu", function () {
  it("should render", function () {
    const { container } = render(
      <MemoryRouter>
        <FinancialHeaderNavItem {...defaultProps} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("should add class to active route", function () {
    const { container } = render(
      <MemoryRouter initialEntries={["/financial/overview"]}>
        <FinancialHeaderNavItem {...defaultProps} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
