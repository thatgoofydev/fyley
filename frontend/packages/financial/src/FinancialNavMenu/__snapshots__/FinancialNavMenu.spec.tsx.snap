// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialNavMenu should add class to active route 1`] = `
<div>
  <li
    class="HeaderNavItem-module_navItem__qJl0a HeaderNavItem-module_open__528yj"
  >
    <span
      class="HeaderNavItem-module_labelWithSub__2BcOK HeaderNavItem-module_active__1fUqf"
    >
      Financial
    </span>
    <ul
      class="HeaderNavItem-module_subNav__3MoHr"
    >
      <li
        class="HeaderNavItem-module_navItem__qJl0a"
      >
        <span
          class=""
        >
          overview
        </span>
      </li>
      <li
        class="HeaderNavItem-module_navItem__qJl0a"
      >
        <span
          class=""
        >
          accounts
        </span>
      </li>
    </ul>
  </li>
</div>
`;

exports[`FinancialNavMenu should render 1`] = `
<div>
  <li
    class="HeaderNavItem-module_navItem__qJl0a HeaderNavItem-module_open__528yj"
  >
    <span
      class="HeaderNavItem-module_labelWithSub__2BcOK"
    >
      Financial
    </span>
    <ul
      class="HeaderNavItem-module_subNav__3MoHr"
    >
      <li
        class="HeaderNavItem-module_navItem__qJl0a"
      >
        <span
          class=""
        >
          overview
        </span>
      </li>
      <li
        class="HeaderNavItem-module_navItem__qJl0a"
      >
        <span
          class=""
        >
          accounts
        </span>
      </li>
    </ul>
  </li>
</div>
`;
